src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -38,6 +38,32 @@
   (fn [request]
     (ring-response/content-type (handler request) "text/plain")))
 
+(defn- add-query-param
+  "Given a query string, add the given key and value to it. If the given
+  query string is empty or nil then the resulting string will be the single
+  KEY=VALUE expression."
+  [query-str key value]
+  {:pre [(or (string? query-str) (nil? query-str))
+         (or (string? key) (keyword? key))
+         (string? value)]
+   :post [(string? %)]}
+  (let [param-pairs (if (str/blank? query-str)
+                      {}
+                      (ring-codec/form-decode query-str))]
+    (ring-codec/form-encode (assoc param-pairs key value))))
+
+(defn- add-source-permissions-param
+  "Add the source_permissions header to the request. This is needed since
+  this was considered a default in Puppet 3.x and is now turned off by default
+  in Puppet 4.x. See SERVER-684."
+  [handler]
+  (fn [request]
+    (let [{:keys [query-string]} request
+          updated-request (assoc request :query-string
+                            (add-query-param
```

Comment:
Yeah, that's actually a much better idea in general. This comment has also pointed out that my docstring is incorrect. I'll get this fixed.